src/MeasureToolbox/integrals.jl

```diff
@@ -71,77 +79,153 @@ This is not compatible with individual dependent parameters. Contains no fields.
+    GaussJacobi <: FiniteGaussQuad
 
 An integral evaulation method that uses Gauss-Jacobi quadrature to
-evaluate integrals. This is valid for finite integral domains. This requires the user
+evaluate integrals. It will take this form:
+
+``\\int_{-1}^{1} f(x) (1-x)^\\alpha (1+x)^\\beta dx \\approx \\sum_{i=1}^{n} \\alpha_i f(x_i)``
+
+Where, 
+
+`` (1-x)^\\alpha (1+x)^\\beta`` 
+
+is the weight function.
+This is valid for finite integral domains. This requires the user
 to input the alpha and beta shape parameters for their function. This will then
 generate its own set of supports and will ignore other parameter supports.
 This is not compatible with individual dependent parameters.
+If α or β is < -1, an error will be returned.  
 
 **Fields**
-- `α::Float64`: The α Jacobi shape parameter.
-- `β::Float64`: The β Jacobi shape parameter.
+- `α::Float64`: Shape parameter that must be > -1
+- `β::Float64`: Shape parameter that must be > -1
 """
-struct GaussJacobi <: AbstractUnivariateMethod
+struct GaussJacobi <: FiniteGaussQuad
     α::Float64
     β::Float64
-    function GaussJacobi(α::Real, β::Real)
-        return new(Float64(α), Float64(β))
+    function GaussJacobi(α::Float64, β::Float64)
+        if α < -1 || β < -1
+            error("α and β must be greater than -1")
+        end
+        return new(α, β)
+    end
+end
+
+"""
+    GaussChebyshev <: FiniteGaussQuad
+
+An integral evaulation method that uses Gauss-Chebyshev quadrature to
+evaluate integrals. This is valid for finite integral domains. This requires the user
+to input the order of Guass-Chebyshev Quadrature they want to use. 
+If the order is not between 1 and 4 an error will be returned. 
+The integral evaluated is as follows:
+
+``\\int_{-1}^{1} f(x) w(x) \\approx \\sum_{i=1}^{n} \\alpha_i f(x_i)``
+
+The weight functions are as follows: 
+
+- 1st order: ``w(x)  =  \\frac{1}{\\sqrt{1-x^2}}``
+- 2nd order: ``w(x) = {\\sqrt{1-x^2}}``
+- 3rd order: ``w(x) = \\sqrt{(1+x)/(1-x)}``
+- 4th order: ``w(x) = \\sqrt{(1-x)/(1+x)}``
+
+This will then generate its own set of supports and will ignore other parameter supports.
+This is not compatible with individual dependent parameters.
+
+**Fields**
+- `order::Int64`: Specifies the order of Gauss-Chebyshev Quadrature. Must be between 1 and 4.
+"""
+struct GaussChebyshev <: FiniteGaussQuad
+    order::Int64
+    function GaussChebyshev(order::Int64)
+        if order < 1 || order > 4
+            error("Order must be between 1 and 4")
+        end
+        return new(order)
     end
 end
 
 """
     GaussLaguerre <: AbstractUnivariateMethod
 
 An integral evaulation method that uses Gauss-Laguerre quadrature to
-evaluate integrals. This is valid for semi-infinite integral domains. Note this
-will generate its own set of supports and will ignore other parameter supports.
-This is not compatible with individual dependent parameters. Contains no fields.
+evaluate integrals. This is valid for semi-infinite integral domains. 
```

Comment:
We should make a note that this supports semi-infinite domains in either direction (we reformulate as needed)